### DISCUSSION TITLE
Statistics reset after configured period

### DISCUSSION THREAD

**Discussion by snothub** [AdguardTeam/AdGuardHome#6356] [Q&A]
I've been running AGH on a raspbian for years and now I just configured a new instance on a router running openwrt. I have query logs rotation set to 30 days and statistics to 7, just as I had earlier. However, now it seems statistics reset completely after 7 days and start over, while previously I'm pretty sure I was always seeing stats for the last 7 days at any point.

Can anyone help me figure this out? Is it supposed to reset like that?

**Comment by ainar-g** [comment]
Is your question about statistics or the query log?  Because these are two different features.

In any case, the query log rotation period tells AGH when it should rename the older log to `querylog.json.1` and start a new `querylog.json`.  Normally, AGH should be able to fetch query log data from both files, but your setup must allow it to access the file.

**Reply by snothub** [reply]
Thanks, my question is why stats are being reset after 7 days given my configuration. I know these are two different things and query log is set to rotate on 30 days and seems to be doing ok. But the stats are set to 7 days and I don't get why it resets to blank after that time.

**Reply by ainar-g** [reply]
I see, thanks.  That obviously shouldn't be happening, but unless something removing `data/stats.db` every week, I'm not sure what could be happening.  As a start, you could inspect the logs and specifically look for lines with the string `stats:` in them, as well as any `[error]`s.

Also, just in case.  Are you using the official builds or the OpenWrt package?  If I recall correctly, the OpenWrt community is supporting their own build, which may differ from the official one.

**Reply by snothub** [reply]
I've also added some more info in [this](https://github.com/AdguardTeam/AdGuardHome/discussions/5272#discussioncomment-7407114) thread. It might be what's causing the reset, not the time period. I'm still trying to figure it out.
